src/lib/utils/assetValue.ts

```diff
@@ -14,12 +14,15 @@ import type {
 
 import { toToken } from "./formatter";
 
-export const calculateAssetValue = (
+const calculateAssetValue = (
   amount: Token<BigSource>,
   price: USD<number>
 ): USD<Big> => big(amount).mul(price) as USD<Big>;
 
 export const calAssetValueWithPrecision = (balance: Balance): USD<Big> => {
+  if (Number.isNaN(Number(balance.amount)) || !balance.amount.length)
```

Comment:
Please add the `trim()` function and also introduce a test case for when `balance.amount` is a space.